src/02-common-principles.md

```diff
@@ -55,6 +56,39 @@ misunderstanding we clarify them here.
     due to different subject response or randomized nature of the stimuli). Run
     is a synonym of a data acquisition.
 
+## File names consist of key-value pairs (entities)
+
+We define file names with a chain of key-value pairs, each of which we call
+an *entity*. Two prominent examples are the `subject` and `session` entities.
+
+If we want to name a datafile that was collected in a certain `session` from
+a certain `subject`, the name of that datafile will start with the following
+string: `sub-<label>_ses-<label>`.
+
+Note that `sub-<sub-label>` correspondes to the `subject` entity because it has
+the `sub-` "key" and`<sub-label>` "value", where `<sub-label>` would in a real
+datafile correspond to a unique identifier of that subject, such as `01`. The
+same  explanation holds for the `session` entity with its `ses-` key and its
+`<ses-label>` value.
+
+Finally, you can notice how different entities are separated by an underscore
+(`_`).
+
+With this system we can name datafiles in a human readable way:
+`sub-01_task-rest_eeg.edf`
+
+From the file name alone, a user can tell that the file contains resting state
+data from subject `01`. The *suffix* `_eeg` and the extension `.edf` are
```

Comment:
Not sure about this one, but this puts "suffix" and "extension" on visual parity. Feel free to reject if you have a principled reason not to treat them the same.

```suggestion
data from subject `01`. The *suffix* `_eeg` and the *extension* `.edf` are
```